### PR TITLE
Documentation: Native testing, buildtime and runtime profile

### LINKED ISSUE #25519: @QuarkusIntegrationTest starts native executable in prod profile instead of test profile
### Describe the bug

I was struggling with database test data population during native-image test in a project and it boiled down to the simple fact that the native-executable is started in *prod* profile (hence expecting `%prod` values) instead of the *test* profile the JVM test correctly uses. 

### Expected behavior

native-image executable is started in *test* profile during test execution.

### Actual behavior

native-image executable is started in *prod* profile during test execution.

### How to Reproduce?

I reproduced it with this tiny example downloaded from code.quarkus.io:
```
$ curl -O -J  https://code.quarkus.io/d?e=io.quarkus:quarkus-resteasy
$ unzip code-with-quarkus.zip 
$ cd code-with-quarkus 
```
Patch with this patch:
```diff
diff --git a/src/main/java/org/acme/GreetingResource.java b/src/main/java/org/acme/GreetingResource.java
index 055f53c..b4e1e1e 100644
--- a/src/main/java/org/acme/GreetingResource.java
+++ b/src/main/java/org/acme/GreetingResource.java
@@ -1,5 +1,7 @@
 package org.acme;
 
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
@@ -8,9 +10,12 @@ import javax.ws.rs.core.MediaType;
 @Path("/hello")
 public class GreetingResource {
 
+    @ConfigProperty(name = "my.property")
+    String property;
+
     @GET
     @Produces(MediaType.TEXT_PLAIN)
     public String hello() {
-        return "Hello RESTEasy";
+        return property;
     }
 }
diff --git a/src/main/resources/application.properties b/src/main/resources/application.properties
index e69de29..7f9990a 100644
--- a/src/main/resources/application.properties
+++ b/src/main/resources/application.properties
@@ -0,0 +1,2 @@
+%test.my.property=Hello, test!
+my.property=Hello.
diff --git a/src/test/java/org/acme/GreetingResourceTest.java b/src/test/java/org/acme/GreetingResourceTest.java
index 87fbd5c..e58f732 100644
--- a/src/test/java/org/acme/GreetingResourceTest.java
+++ b/src/test/java/org/acme/GreetingResourceTest.java
@@ -15,7 +15,7 @@ public class GreetingResourceTest {
           .when().get("/hello")
           .then()
              .statusCode(200)
-             .body(is("Hello RESTEasy"));
+             .body(is("Hello, test!"));
     }
 
 }
```

Running the tests as:
```
./mvnw  verify -Pnative
```
Passes the JVM mode:
```
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.953 s - in org.acme.GreetingResourceTest
```
And fails with native:
```
[ERROR] org.acme.GreetingResourceIT.testHelloEndpoint  Time elapsed: 1.452 s  <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Response body doesn't match expectation.
Expected: is "Hello, test!"
  Actual: Hello.
```
Note how the Quarkus app starts:

```
Executing "/tmp/code-with-quarkus/target/code-with-quarkus-1.0.0-SNAPSHOT-runner -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/tmp/code-with-quarkus/target/quarkus.log -Dquarkus.log.file.enable=true"
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
INFO  [io.quarkus] (main) code-with-quarkus 1.0.0-SNAPSHOT native (powered by Quarkus 2.9.0.Final) started in 0.020s. Listening on: http://0.0.0.0:8081
INFO  [io.quarkus] (main) Profile prod activated. 
INFO  [io.quarkus] (main) Installed features: [cdi, resteasy, smallrye-context-propagation, vertx]
```
i.e. the profile should be **test**, not **prod**. 

### Output of `uname -a` or `ver`

Linux amd64

### Output of `java -version`

OpenJDK Runtime Environment Temurin-17.0.3+7 (build 17.0.3+7)

### GraalVM version (if different from Java)

native-image 22.1.0.0-Final Mandrel Distribution (Java Version 17.0.3+7)

### Quarkus version or git rev

2.9.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Karm**
fixes #25519

Built doc available here: [https://karms.biz/pastebin/Building-a-Native-Executable.html](https://karms.biz/pastebin/Building-a-Native-Executable.html#profiles)

Original: [https://quarkus.io/guides/building-native-image](https://quarkus.io/guides/building-native-image#excluding-tests-when-running-as-a-native-executable)

### CODE REVIEW

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 1]
```diff
@@ -271,12 +271,38 @@ This procedure was formerly accomplished using the `@NativeImageTest` annotation
...
 ====
 
-By default, integration tests runs using the `prod` profile.
-This can be overridden using the `quarkus.test.native-image-profile` property.
-For example, in your `application.properties` file, add: `quarkus.test.native-image-profile=test`.
-Alternatively, you can run your tests with: `./mvnw verify -Pnative -Dquarkus.test.native-image-profile=test`.
-However, don't forget that when the native executable is built the `prod` profile is enabled.
-So, the profile you enable this way must be compatible with the produced executable.
+=== Profiles
+By default, integration tests both `build` and `run` the native executable using the `prod` profile.
```
Comment (geoand) [inline docs/src/main/asciidoc/building-native-image.adoc:275]:
```suggestion
By default, integration tests both *build* and *run* the native executable using the `prod` profile.
```

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 2]
```diff
@@ -271,12 +271,38 @@ This procedure was formerly accomplished using the `@NativeImageTest` annotation
...
+
+You can override the profile the executable `runs` with during the test using the `quarkus.test.native-image-profile` property.
```
Comment (geoand) [inline docs/src/main/asciidoc/building-native-image.adoc:277]:
```suggestion
You can override the profile the executable *runs* with during the test using the `quarkus.test.native-image-profile` property.
```

**Review by geoand — COMMENTED** [review]
Looks good! Just added some tiny suggestions

**Comment by Karm** [comment]
@geoand Ack. Updated.

See: https://karms.biz/pastebin/Building-a-Native-Executable.html#profiles

**Review by geoand — APPROVED** [review]

**Pull request merged**